tests/e2e/test_triton.py

```diff
@@ -0,0 +1,234 @@
+
+
+def test_triton_multiple_funcs_and_multiple_calls():
+    @triton.jit
+    def kernel1(x_ptr):
+        x = tl.load(x_ptr)
+        tl.store(x_ptr, x + 1.0)
+
+    @triton.jit
+    def kernel2(x_ptr):
+        x = tl.load(x_ptr)
+        tl.store(x_ptr, x + 2.0)
+
+    ArrayType = MemRefFactory((DYNAMIC,), F32)
+
+    @compile()
+    def func_test(x: ArrayType):
+        kernel1(x)
+        kernel2(x)
+        kernel1(x)
+
+    mlir = func_test.emit_mlir()
+    assert mlir.count("func.func @kernel1") == 1
+    assert "func.func @kernel2" in mlir
+    x = np.zeros(1, dtype=np.float32)
+    func_test(x)
+    assert x[0] == 4.0
+
+
+def test_triton_different_signatures():
+    @triton.jit
+    def kernel(a_ptr, out_ptr):
+        a = tl.load(a_ptr)
+        tl.store(out_ptr, a)
+
+    int_arr = MemRefFactory((DYNAMIC,), UInt32)
+    fp_arr = MemRefFactory((DYNAMIC,), F32)
+
+    @compile()
+    def func_test(
+        m_int: int_arr, out_int: int_arr, m_fp: fp_arr, out_fp: fp_arr
+    ):
+        kernel(m_int, out_int)
+        kernel(m_fp, out_fp)
+
+    np.random.seed(0)
+    m_int = np.random.rand(1).astype(np.uint32)
+    out_int = np.zeros(1, dtype=np.uint32)
+    m_fp = np.random.rand(1).astype(np.float32)
+    out_fp = np.zeros(1, dtype=np.float32)
+    func_test(m_int, out_int, m_fp, out_fp)
+    assert m_int[0] == out_int[0]
+    assert m_fp[0] == out_fp[0]
+
+
+def test_triton_unused_args():
+    @triton.jit
+    def kernel(x_ptr, y_ptr, out_ptr, a, BLOCK_SIZE: tl.constexpr, b):
+        pid = tl.program_id(axis=0)
+        x = tl.load(x_ptr + pid)
+        tl.store(out_ptr + pid, x + a)
+
+    ArrayType = MemRefFactory((DYNAMIC,), F32)
+
+    @compile()
+    def func_test(x: ArrayType, y: ArrayType, out: ArrayType, size: Index):
+        a: UInt32 = 3
+        BLOCK_SIZE: Number = 64
+        b: UInt32 = 1
+        for i in arange(size):
+            kernel(x, y, out, a, BLOCK_SIZE, b, i)
+
+    size = 3
+    x = np.zeros(size, dtype=np.float32)
+    y = np.zeros(size, dtype=np.float32)
+    out = np.zeros(size, dtype=np.float32)
+    func_test(x, y, out, size)
+    assert np.allclose(out, x + 3)
+
+
+def test_triton_inlined_args():
+    @triton.jit
+    def kernel(x_ptr, out_ptr, a, BLOCK_SIZE: tl.constexpr):
+        pid = tl.program_id(axis=0)
+        block_start = pid * BLOCK_SIZE
+        offsets = block_start + tl.arange(0, BLOCK_SIZE)
+        x = tl.load(x_ptr + offsets)
+        tl.store(out_ptr + offsets, x + a)
+
+    ArrayType = MemRefFactory((DYNAMIC,), F32)
+
+    @compile()
+    def func_test(x: ArrayType, out: ArrayType, size: Index):
+        for i in arange(size // 64):  # size // BLOCK_SIZE
+            kernel(x, out, UInt32(3), 64, i)
+        for i in arange(size // 128):
+            kernel(out, out, UInt32(15), 128, i)
+
+    size = 65536
+    x = np.zeros(size, dtype=np.float32)
+    out = np.zeros(size, dtype=np.float32)
+    func_test(x, out, size)
+    assert np.allclose(out, x + 18)
```

Comment:
```suggestion
    x = multi_arange((size,), np.float32)
    out = np.zeros(size, dtype=np.float32)
    func_test(x, out, size)
    assert np.allclose(out, x + 3 + 15)
```
Since I a made other comments anyway, it's probably marginally better if not all elements of x are the same.